src/Message/Analytics/SessionCreatedMessage.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Message\Analytics;
+
+use App\Message\Analytics\Helper\AnalyticsDataType;
+use App\Message\Analytics\Helper\GameConfigAnalyticsHelper;
+use App\Message\Analytics\Helper\GameSessionAnalyticsHelper;
+use DateTimeImmutable;
+use JsonSerializable;
+use Symfony\Component\Uid\Uuid;
+
+class SessionCreatedMessage extends AnalyticsMessageBase implements JsonSerializable
+{
+    public readonly GameSessionAnalyticsHelper $session;
+    public readonly GameConfigAnalyticsHelper $config;
+    public readonly string $userName;
+    public readonly int $accountId;
+
+    public function __construct(
+        DateTimeImmutable $timeStamp,
+        Uuid $serverManagerId,
+        string $userName,
+        int $accountId,
+        int $sessionId,
+        string $sessionName,
+        DateTimeImmutable $gameCreationTime,
+        int $gameStartYear,
+        int $gameEndMonth,
+        string $configFileName,
+        string $configVersion,
+        string $configVersionMessage,
+        string $configRegion,
+        string $configDescription
+    ) {
+        parent::__construct(
+            new AnalyticsDataType(AnalyticsDataType::GAME_SESSION_CREATED),
+            $timeStamp,
+            $serverManagerId
+        );
+        $this->userName = $userName;
```

Comment:
As an example, you can use property promotion. 

"Using a (public, protected or private) is the most likely way to apply property promotion, but any other single modifier (such as readonly) will have the same effect."

so I removed this
public readonly string $userName;
to  constructor argument. And then you do not need any assigments in the constructor body

```suggestion
    public readonly GameSessionAnalyticsHelper $session;
    public readonly GameConfigAnalyticsHelper $config;
    public readonly int $accountId;

    public function __construct(
        DateTimeImmutable $timeStamp,
        Uuid $serverManagerId,
        public readonly string $userName,
        int $accountId,
        int $sessionId,
        string $sessionName,
        DateTimeImmutable $gameCreationTime,
        int $gameStartYear,
        int $gameEndMonth,
        string $configFileName,
        string $configVersion,
        string $configVersionMessage,
        string $configRegion,
        string $configDescription
    ) {
        parent::__construct(
            new AnalyticsDataType(AnalyticsDataType::GAME_SESSION_CREATED),
            $timeStamp,
            $serverManagerId
        );
```

Reply:
I did learn about this and know how it works. The reason I chose to not use it is because to me it seemed clearer to be able to check what the member variables of a class are. However, if the convention is to use this method of adding member variables to classes, im happy to follow that.